main/src/ca/uwaterloo/flix/language/phase/CodegenHelper.scala

```diff
@@ -191,6 +249,40 @@ object CodegenHelper {
   }
 
   /**
+    * At this method, we group `tuples` that have the same field representation so we only generate one class for them.
+    * If a field is a primitive, then it can be represented by it's primitive but if the field is not a primitive then it
+    * has to be represented using an object.
+    * For example, `(List[Int32], Bool)` and `(Result[Int32,Int32], Bool)` have the same representation since the first field
+    * of both of them is an object and the second field is a `Bool`. So we only create one tuple class for both of these
+    * tuples.
+    */
+  def groupTuplesByFieldTypes(tuples: List[Type]): List[List[List[Type]]] = tuples.map{
```

Comment:
So the signature here is hard to understand, triple nested lists?

What I think need is a representation of Java types then you can map one Flix type into its Java type, and if you have a set of Flix types you can map them (a possible smaller) set of Java types (which is really what is happening here).

Reply:
The comment was helpful though.